app/src/main/kotlin/co/touchlab/droidconandroid/UserDetailFragment.kt

```diff
@@ -67,32 +76,28 @@ class UserDetailFragment : Fragment() {
     }
 
     override fun onDestroy() {
+        viewModel.unregister()
         super.onDestroy()
-        EventBusExt.getDefault().unregister(this)
     }
 
-    private fun findUserCodeArg(): String {
-        var userCode = arguments?.getString(USER_CODE)
-        if (userCode.isNullOrBlank()) {
-            userCode = activity.intent.getStringExtra(USER_CODE)
-        }
-
-        if (userCode.isNullOrBlank())
-            throw IllegalArgumentException("Must set user code")
+    private fun findUserId(): Long {
+        val userId = activity.intent.getLongExtra(UserDetailActivity.USER_ID, 0L)
+        if (userId == 0L)
+            throw IllegalArgumentException("Must set user id")
```

Comment:
This is what caused a crash when we were using usercode. Instead of throwing an exception, can we call findUserError(message: String) to Log, toast, and close activity?